frontend/src/metabase/collections/components/CollectionSidebar/Bookmarks/getIcon.ts

```diff
@@ -0,0 +1,18 @@
+import { Bookmark } from "metabase-types/api";
+
+export function getIcon({ authority_level, display, type }: Bookmark) {
+  if (display) {
+    return display;
+  }
+
+  if (type === "collection") {
+    return authority_level === "official" ? "badge" : "folder";
+  }
+
+  const icons = {
+    card: "grid",
+    dashboard: "dashboard",
+  };
```

Comment:
Can we use the entities' `getIcon` functions here?

First of all, in this way we have a single source of truth for entity icons. So if we want to, let's say, change the dashboard icon, we can only patch the dashboard entity's `getIcon` method and this will be enough. With these changes, we'll now need to keep in mind that bookmarks have to be changed separetely.

Secondly, these methods encapsulate business logic too. For instance, the official collection's icon should _only_ be displayed when running a paid Metabase plan. So if you had a paid instance, created a few official collections, and then downgraded to the free version, the FE should not display official icons anymore. With these changes, the icon is actually going to be displayed in free versions too. On the other hand, [collection entity's `getIcon` method actually handles that](https://github.com/metabase/metabase/blob/b9602db06c0568e608357e719cce20ef38e7212e/frontend/src/metabase/entities/collections.js#L89).